File: 75 Sort Colors/75.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if not nums or len(nums) <= 1:
            return
        
        def swap(i , j):
            nums[i], nums[j] = nums[j], nums[i]

        i = 0
        cur = 0
        k = len(nums) -1 
        while cur <= k:
            if nums[cur] == 0:
                swap(i , cur)
                i += 1
                cur += 1
            elif nums[cur] == 1:
                cur += 1
            else:
                swap(k , cur)
                k -= 1

File: 75 Sort Colors/75.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    if (!nums || nums.length == 1) {
        return;
    }
    let swap = (a, b) => ([nums[a], nums[b]] = [nums[b], nums[a]]);
    let i = 0, cur = 0, k = nums.length-1;

    while (cur <= k) {
        if (nums[cur] == 0) {
            swap(i++, cur++)
        } else if (nums[cur] == 1) {
            cur++
        } else {
            swap(cur, k--)
        }
    }
};
